### ISSUE TITLE
Process user input

### ISSUE THREAD

**Issue by yagueto** [Yagueteiro/code-jam-2021#2]
We need to check whether the user has guessed correctly. Even though the website gives us some descriptions of the image, the user may submit a synonym or something similar.

**Comment by JesperDramsch** [comment]
I'd love to take a shot at this later today.

**Comment by JesperDramsch** [comment]
The code in user_input can now retrieve synonyms and validate guesses with typos.

**Comment by yagueto** [comment]
I've just been checking everything and I have a small issue: when I type
 ```python
>>> validate("feline", synonyms=get_synonyms("cat"))
``` 
it returns true as expected. However, if I type 
 ```python
>>> validate("f", synonyms=get_synonyms("cat"))
``` 
the answer is True again. I've also tried changing the threshold, but it doesn't seem to change the return value

**Comment by JesperDramsch** [comment]
> I've just been checking everything and I have a small issue: when I type
> [...]
> the answer is True again. I've also tried changing the threshold, but it doesn't seem to change the return value

I made a wrong assumption that we can take partial matches. Fixed in #10
